src/sage/graphs/graph_decompositions/modular_decomposition.pyx

```diff
@@ -638,19 +605,82 @@ def habib_maurer_algorithm(graph, g_classes=None):
                          for sg in d1.values()]
         return root
 
-    root = create_series_node()
+    root = Node(NodeType.SERIES)
     root.children = [habib_maurer_algorithm(graph.subgraph(vertices=sg), g_classes)
                      for sg in g_comp.connected_components(sort=False)]
     return root
 
 
-modular_decomposition = habib_maurer_algorithm
+################################################################################
+#                   Exported modular_decomposition function                    #
+################################################################################
+def modular_decomposition(G, algorithm=None):
+    r"""
+    Return the modular decomposition of the current graph.
+
+    This function should not be used directly, it should be called via the
+    ``modular_decomposition`` method of ``Graph``.
+
+    TESTS::
```

Comment:
add `INPUT:` block

Reply:
done in [451b325](https://github.com/sagemath/sage/pull/39038/commits/451b3256ac2cbbb00d2ff24ecc80339fe73d5e40)